src/System.Net.Http/src/System/Net/Http/Headers/HeaderUtilities.cs

```diff
@@ -249,14 +249,69 @@ internal static void CheckValidQuotedString(string value, string parameterName)
             return null;
         }
 
-        internal static bool TryParseInt32(string value, out int result)
+        internal static bool TryParseInt32(string value, out int result) =>
+            TryParseInt32(value, 0, value.Length, out result);
+
+        internal static bool TryParseInt32(string value, int offset, int length, out int result)
         {
-            return int.TryParse(value, NumberStyles.None, NumberFormatInfo.InvariantInfo, out result);
+            // TODO #21281: Replace with int.TryParse(Span<char>) once it's available
+
+            if (offset < 0 || length < 0 || offset > value.Length - length)
+            {
+                result = 0;
+                return false;
+            }
+
+            int tmpResult = 0;
+            int pos = offset;
+            for (int i = 0; i < length; i++)
```

Comment:
It doesn't look like you use 'i' in the loop, so why not make the loop cond something like "pos < end" where end = offset + length?